### PR TITLE
Allow facets in multi-valued objects and other improvements

### PR DESCRIPTION
**Pull request by Cito**
The PR contains these fixes and improvements:
- allow facets in multi-valued objects and test this
- always return the facet options sorted by value
- do not change the config in place when the ID is missing as selected field
  - this caused the response of search-options to change after the first query
- replace dots with blanks when auto creating names from keys (so "study.title" becomes "Study Title" now)
- more than double the test speed by reusing state

### CODE REVIEW

File: tests/fixtures/joint.py [hunk 1]
```diff
@@ -38,52 +38,89 @@
 QueryParams: TypeAlias = Mapping[str, int | str | list[str]]
 
 
+@dataclass
+class State:
+    """The current state of the database and the event topics"""
+
+    database_dirty: bool
+    events_dirty: bool
```
Comment (TheByronHimes) [inline tests/fixtures/joint.py:46]: It doesn't look like `events_dirty` is getting used.

**Review by TheByronHimes — APPROVED** [review]
Nicely done

File: tests/fixtures/joint.py (hunk 1 above)
Comment (Cito) [reply tests/fixtures/joint.py:46]: It's used only once in `reset_state`.

Comment (Cito) [reply tests/fixtures/joint.py:46]: Will be solved properly in #31.

**Pull request merged**